app/src/main/java/com/ddoczi/tasky/agenda/data/remote/worker/SyncRemoteWithLocalWorker.kt

```diff
@@ -0,0 +1,107 @@
+import com.ddoczi.tasky.agenda.domain.model.SyncType
+import com.ddoczi.tasky.agenda.domain.repository.AgendaRepository
+import com.ddoczi.tasky.agenda.domain.repository.EventRepository
+import com.ddoczi.tasky.agenda.domain.repository.ReminderRepository
+import com.ddoczi.tasky.agenda.domain.repository.TaskRepository
+import com.ddoczi.tasky.agenda.enums.AgendaType
+import dagger.assisted.Assisted
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.supervisorScope
+
+@HiltWorker
+class SyncRemoteWithLocalWorker(
+    @Assisted val context: Context,
+    @Assisted val workerParameters: WorkerParameters,
+    private val eventRepository: EventRepository,
+    private val taskRepository: TaskRepository,
+    private val reminderRepository: ReminderRepository,
+    private val agendaRepository: AgendaRepository,
+    private val eventUploader: EventUploaderImpl
+) : CoroutineWorker(context, workerParameters) {
+    override suspend fun doWork(): Result {
+        val items = agendaRepository.getAllSyncableItems()
+
+        return try {
+            supervisorScope {
+                val jobs = items.map { launch { completeAction(it) } }
+                jobs.forEach { it.join() }
+            }
+            Result.success()
+        } catch (e: Exception) {
+            Result.retry()
+        }
+    }
+
+    private suspend fun completeAction(syncItem: SyncItem) {
+        if (syncItem.type == SyncType.DELETE) {
+            if (deleteItem(syncItem) == Result.success()) {
+                agendaRepository.deleteSyncableItem(syncItem.id)
+            }
+            return
+        }
+
+        val item = when (syncItem.agendaItemType) {
+            AgendaType.EVENT -> eventRepository.getEventById(syncItem.id)
+            AgendaType.REMINDER -> reminderRepository.getReminderById(syncItem.id)
+            AgendaType.TASK -> taskRepository.getTaskById(syncItem.id)
+        }
+        val runner = when (syncItem.type) {
+            SyncType.CREATE -> createItem(item)
+            SyncType.UPDATE -> updateItem(item)
+            else -> Result.success()
+        }
+
+        if(runner == Result.success()) {
+            agendaRepository.deleteSyncableItem(syncItem.id)
+        }
+
+    }
+
+    private suspend fun deleteItem(syncItem: SyncItem): Result {
+        return try {
+            when (syncItem.agendaItemType) {
+                AgendaType.EVENT -> eventRepository.deleteEventById(syncItem.id)
+                AgendaType.REMINDER -> reminderRepository.deleteReminderById(syncItem.id)
+                AgendaType.TASK -> taskRepository.deleteTaskById(syncItem.id)
+            }
+            Result.success()
+        } catch (e: Exception) {
+            Result.failure()
+        }
+    }
+
+    private suspend fun updateItem(agendaItem: AgendaItem): Result {
+        return try {
+            when (agendaItem) {
+                is AgendaItem.Event -> eventUploader.uploadEvent(agendaItem, true)
```

Comment:
Here, I'd be a fan of using named parameters. You get IDE inline hints, but for me as the reviewer I have no idea what this `true` stands for. I pretty much always stick to named parameters because of that, makes the code a lot more readable